projects/bp-gallery/src/components/views/exhibitions/ExhibitonUtils.tsx

```diff
@@ -444,23 +430,25 @@ const ExhibitionDragNDrop = ({
 }>) => {
   const [isSorting, setIsSorting] = useState(false);
 
-  const swapSectionDraggables = (oldIndex: number, newIndex: number, sectionId: string) => {
-    databaseSaver.swapOrderInExhibitionPicture(sections, oldIndex, newIndex, sectionId);
-    setSections(
-      sections.map(section =>
-        section.id === sectionId
-          ? ({
-              id: section.id,
-              text: section.text,
-              dragElements: section.dragElements.map((elem, index) => {
-                if (index === oldIndex) return section.dragElements[newIndex];
-                if (index === newIndex) return section.dragElements[oldIndex];
-                return elem;
-              }),
-            } as SectionState)
-          : section
-      )
+  const moveSections = (oldSectionId: string, newSectionId: string) => {
+    const oldIndex = sections.indexOf(sections.find(section => section.id === oldSectionId)!);
+    const newIndex = sections.indexOf(sections.find(section => section.id === newSectionId)!);
+    const newSectionOrder = arrayMove(sections, oldIndex, newIndex);
+    databaseSaver.updateSectionsOrder(newSectionOrder);
+    setSections(newSectionOrder);
+  };
+  const deleteSection = (sectionId: string) => {
+    databaseSaver.deleteSection(sectionId);
+    setSections(sections => sections.filter(section => section.id !== sectionId));
+  };
+  const moveSectionDraggables = (oldIndex: number, newIndex: number, sectionId: string) => {
+    const newDraggablesOrder = sections.map(section =>
+      section.id === sectionId
+        ? { ...section, dragElements: arrayMove(section.dragElements, oldIndex, newIndex) }
+        : section
     );
+    setSections(newDraggablesOrder);
+    databaseSaver.moveOrderInExhibitionPicture(sections, sectionId);
```

Comment:
`setSections` is not synchronous, so the line does not take effect immediatly, thus we have to pass `newDraggablesOrder` instead of `sections`, otherwise the database will always lag one change behind.
```suggestion
    databaseSaver.moveOrderInExhibitionPicture(newDraggablesOrder, sectionId);
```